### PR TITLE
Hook up compression in WebSocketReader and WebSocketWriter

### PR DESCRIPTION
**Pull request by swankjesse**
It isn't yet enabled in the calling RealWebSocket, so this is only
visible to tests right now.

### CODE REVIEW

**Comment by swankjesse** [comment]
FYI @AntonRutkevich

File: okhttp/src/main/java/okhttp3/internal/ws/WebSocketWriter.kt [hunk 1]
```diff
@@ -39,11 +41,15 @@ import okio.ByteString
...
+class WebSocketWriter(
   private val isClient: Boolean,
   val sink: BufferedSink,
-  val random: Random
-) {
+  val random: Random,
+  private val messageDeflater: MessageDeflater?,
+  private val minimumDeflateSize: Long
+) : Closeable {
+  /** This buffer is holds outbound data for compression and masking. */
```
Comment (yschimke) [inline okhttp/src/main/java/okhttp3/internal/ws/WebSocketWriter.kt]: nit: "is hold"

**Comment by yschimke** [comment]
LGTM

**Comment by swankjesse** [comment]
Gonna fix typo in follow-up.

**Comment by AntonRutkevich** [comment]
Much cleaner than the initial version due to previous refactoring. 👍

**Pull request merged**